packages/langium/src/references/scope-provider.ts

```diff
@@ -56,11 +56,11 @@ export class DefaultScopeProvider implements ScopeProvider {
         if (precomputed) {
             let currentNode: AstNode | undefined = context.container;
             do {
-                const allDescriptions = precomputed.get(currentNode);
-                if (allDescriptions.length > 0) {
-                    scopes.push(stream(allDescriptions).filter(
-                        desc => this.reflection.isSubtype(desc.type, referenceType)));
-                }
+                scopes.push(
+                    precomputed.getStream(currentNode).filter(
+                        desc => this.reflection.isSubtype(desc.type, referenceType)
+                    )
+                );
```

Comment:
This creates empty scopes, which was avoided in the previous version. The change might have a performance impact for larger documents. I suggest keeping a check for an empty stream.